uniffi_bindgen/src/interface/mod.rs

```diff
@@ -241,6 +241,22 @@ impl<'ci> ComponentInterface {
         }
     }
 
+    pub fn contains_optional_types(&self) -> bool {
+        self.iter_types()
+            .iter()
+            .any(|t| matches!(t, Type::Optional(_)))
+    }
+
+    pub fn contains_sequence_types(&self) -> bool {
+        self.iter_types()
+            .iter()
+            .any(|t| matches!(t, Type::Sequence(_)))
+    }
+
+    pub fn contains_map_types(&self) -> bool {
+        self.iter_types().iter().any(|t| matches!(t, Type::Map(_)))
+    }
```

Comment:
For completeness, you could consider adding some basic unittests for these methods in the suite at the bottom of this file. I don't expect them to be particularly high-value tests though, the code here is about as "obviously correct" as code ever gets...